packages/next/src/shared/lib/router/adapters.tsx

```diff
@@ -24,13 +24,26 @@ export function adaptForAppRouterInstance(
     },
     fastRefresh() {},
     push(href, { scroll } = {}) {
-      void pagesRouter.push(href, undefined, { scroll })
+      void pagesRouter.push(
+        // TODO: Handle untrusted hrefs
+        typeof href === 'string' ? href : href.unsafeHref.__href,
```

Comment:
This needs validation in development to show warnings

Reply:
I was under the impression that this is the code responsible for the Pages router? I planned on doing the implementation in this PR but wanted to make sure the overall behavior makes sense first.